spdy_test.go

```diff
@@ -252,6 +252,91 @@ func TestHalfClose(t *testing.T) {
 	wg.Wait()
 }
 
+func TestUnexpectedRemoteConnectionClosed(t *testing.T) {
+	tt := []struct {
+		closeReceiver bool
+		closeSender   bool
+	}{
+		{closeReceiver: true, closeSender: false},
+		{closeReceiver: false, closeSender: true},
+		{closeReceiver: false, closeSender: false},
+	}
+	for tix, tc := range tt {
+		listen := "localhost:7743"
```

Comment:
Probably right way to do such things is `localhost:0` and then use `listener.Addr().String()` for `Dial`